lib/libzfs/libzfs_dataset.c

```diff
@@ -2287,7 +2288,19 @@ get_numeric_property(zfs_handle_t *zhp, zfs_prop_t prop, zprop_source_t *src,
 		switch (zfs_prop_get_type(prop)) {
 		case PROP_TYPE_NUMBER:
 		case PROP_TYPE_INDEX:
-			*val = getprop_uint64(zhp, prop, source);
+			/*
+			 * We can directly read createtxg property from zfs
+			 * handle for filesystem and snapshot types.
```

Comment:
Please update the comment as well. Maybe mention that bookmarks specifically are what it doesn't work for. Are there any other types we might have a handle for here?